dietpi/dietpi-software

```diff
@@ -12236,7 +12253,33 @@ _EOF_
 			fi
 
 		fi
+		
+		software_id=185 # Portainer
+		if (( ${aSOFTWARE_INSTALL_STATE[$software_id]} == 1 )); then
+
+			Banner_Configuration
 
+			# Restart Docker service to be able to deploy the container
+			G_DIETPI-NOTIFY 2 'Docker will be restarted to be able to deploy the container.'
+			G_EXEC systemctl daemon-reload
+			G_EXEC systemctl restart docker.service
+			
+			# Check for existing portainer installation
+			if [[ -d '/mnt/dietpi_userdata/docker-data/volumes/portainer_data' ]]; then
+				G_DIETPI-NOTIFY 2 'Portainer detected, container will be dropped before reinstallation.'
+				G_EXEC docker rm "$(docker ps -a | grep "portainer/portainer-ce" | awk '{print $1}')" --force
+				G_EXEC docker rmi "$(docker images -a | grep "portainer/portainer-ce" | awk '{print $3}')"
```

Comment:
`grep` and `awk` can always be merged into a single call. Also, instead of `awk`, we use `mawk` which is the faster Debian default. Only in case it is installed, the slower (but feature-richer) `gawk` will replace the `awk` alternative symlink. So using `mawk` explicitly assures consistency across all systems and faster execution:
```
G_EXEC docker rm "$(docker ps -a | mawk '/portainer\/portainer-ce/{print $1}')" --force
G_EXEC docker rmi "$(docker images -a | mawk '/portainer\/portainer-ce/{print $3}')"
```

Same below in uninstall step

Reply:
ok will change it that way. Let me see if it's working as expected